tests/unit/utils/test_utils.py

```diff
@@ -818,3 +818,44 @@ def test_libgl_workaround(monkeypatch, skip):
     if skip:
         monkeypatch.setenv('QUTE_SKIP_LIBGL_WORKAROUND', '1')
     utils.libgl_workaround()  # Just make sure it doesn't crash.
+
+
+@pytest.mark.parametrize('duration, out', [
+    ("0", 0),
+    ("0s", 0),
+    ("0.5s", 500),
+    ("59s", 59000),
+    ("60", 60),
+    ("60.4s", 60400),
+    ("1m1s", 61000),
+    ("1.5m", 90000),
+    ("1m", 60000),
+    ("1h", 3_600_000),
+    ("0.5h", 1_800_000),
+    ("1h1s", 3_601_000),
+    ("1h 1s", 3_601_000),
+    ("1h1m", 3_660_000),
+    ("1h1m1s", 3_661_000),
+    ("1h1m10s", 3_670_000),
+    ("10h1m10s", 36_070_000),
+])
+def test_parse_duration(duration, out):
+    assert utils.parse_duration(duration) == out
+
+
+@pytest.mark.parametrize('duration, out', [
+    ("-1s", -1),  # No sense to wait for negative seconds
+    ("-1", -1),
+    ("34ss", -1),
```

Comment:
Some more ideas for test values, all of which I'd consider invalid:

- Just `h` without any digit
- `1.1.1s`
- `1.s`
- `.1s`
- `.s`
- Empty string
- `10e5s`
- `5s10m`